src/mp/snapshot.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020 The reone project contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <cstdint>
+#include <set>
+#include <map>
+#include <list>
+#include <memory>
+#include <vector>
+#include <deque>
+
+#include "objects.h"
+#include "../common/streamreader.h"
+#include "../common/streamwriter.h"
+
+namespace reone {
+
+namespace net {
+
+constexpr uint32_t kServerMaxLen = 64;
+constexpr uint32_t kClientMaxLen = 3;  // may increase if too tight
+
+struct Snapshot {
+    Snapshot(uint32_t seq, uint32_t tick) : seq{ seq }, tick{ tick } {};
```

Comment:
Let's use this syntax when variables are constructor initialized: `seq(seq), tick(tick)`